src/rtp/rtp.c

```diff
@@ -333,28 +332,25 @@ static inline int filter_event(struct rtp *session, uint32_t ssrc)
 
 static uint32_t next_csrc(struct rtp *session)
 {
-        /* This returns each source marked "should_advertise_sdes" in turn. */
-        int chain, cc;
-        source *s;
-
-        cc = 0;
-        for (chain = 0; chain < RTP_DB_SIZE; chain++) {
+        int cc = 0;
+        for (int chain = 0; chain < RTP_DB_SIZE; chain++) {
                 /* Check that the linked lists making up the chains in */
                 /* the hash table are correctly linked together...     */
-                for (s = session->db[chain]; s != NULL; s = s->next) {
-                        if (s->should_advertise_sdes) {
-                                if (cc == session->last_advertised_csrc) {
-                                        session->last_advertised_csrc++;
-                                        if (session->last_advertised_csrc ==
-                                            session->csrc_count) {
-                                                session->last_advertised_csrc =
-                                                    0;
-                                        }
-                                        return s->ssrc;
-                                } else {
-                                        cc++;
-                                }
+                for (source *s = session->db[chain]; s != NULL; s = s->next) {
+                        if (!s->should_advertise_sdes)
+                                continue;
+
+                        if (cc != session->last_advertised_csrc) {
+                                cc++;
+                                continue;
                         }
+
+                        session->last_advertised_csrc++;
+                        if (session->last_advertised_csrc == session->csrc_count) {
+                                session->last_advertised_csrc = 0;
+                        }
+                        /* This returns each source marked "should_advertise_sdes" in turn. */
+                        return s->ssrc;
```

Comment:
There are two changes in this block:
1) C99-ization, removing the unneeded forward declarations.
2) Reformulating the inner loop to early-continue and thus move focus towards the real point of the function.

Resolution of clang-tidy's complaints on returns and elses.